web/src/components/HoursMinutes.tsx

```diff
@@ -0,0 +1,155 @@
+import { HStack } from "@chakra-ui/layout";
+import {
+  NumberInput,
+  NumberInputField,
+  InputRightElement,
+  NumberInputStepper,
+  NumberIncrementStepper,
+  NumberDecrementStepper,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+
+type Props = {
+  setValue: (n: number) => void;
+  /** maxDurationSeconds in seconds, this must be set for onBlurFn validation to occur */
+  maxDurationSeconds?: number;
+  /** intitialValue that hydrates the component with hours/mins */
+  initialValue?: number;
+  rightElement?: React.ReactNode;
+};
+
+const HoursMinutes = ({
+  maxDurationSeconds,
+  setValue,
+  initialValue,
+  rightElement,
+}: Props) => {
+  const [hours, setHours] = useState<number>();
+  const [mins, setMins] = useState<number>();
+
+  useEffect(() => {
+    if (
+      initialValue != undefined &&
+      hours === undefined &&
+      mins === undefined
+    ) {
+      setHours(Math.trunc(initialValue / 60 / 60));
+      setMins((initialValue / 60) % 60);
+    }
+  }, [initialValue]);
+
+  let maxH = maxDurationSeconds ? maxDurationSeconds / 3600 : 24;
+
+  const onBlurFn = () => {
+    if (hours != undefined && mins != undefined) {
+      const duration = hours * 60 * 60 + mins * 60;
+      // DE = when an out of bounds value is adjusted to maxSeconds, we need to update the hours and mins to match
+      if (maxDurationSeconds && duration > maxDurationSeconds) {
+        setValue(maxDurationSeconds);
+        let h = Math.trunc(maxDurationSeconds / 60 / 60);
```

Comment:
Floor is actually the better fit here, so changing it
```suggestion
        let h = Math.floor(maxDurationSeconds / 60 / 60);
```